merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MissionModelGenerator.java

```diff
@@ -812,7 +812,7 @@ private Optional<ComputedAttributesCodeBlocks> getComputedAttributesCodeBlocks(
       }
       computedAttributesTypeName = TypeName.get(typeMirror.get());
     } else {
-      effectModelReturnMapperBlock = Optional.of(CodeBlock.of("new $T($T.class)", EnumValueMapper.class, Unit.class));
+      effectModelReturnMapperBlock = Optional.of(CodeBlock.of("$T.$L", BasicValueMappers.class,"$unit()"));
```

Comment:
The fact that this change exists suggests to me that we should use `Resolver` in this branch, too, so that we're *always* consistent with explicitly-declared types.

```java
      effectModelReturnMapperBlock = new Resolver(this.typeUtils, this.elementUtils, missionModel.typeRules)
          .applyRules(new TypePattern.ClassPattern(ClassName.get(Unit.class), List.of()));
```

We do need to make `applyRules` public, but I think that's acceptable.

Reply:
Let me look into that.

Reply:
Got this working now with the below code thanks for all the help: 

```
effectModelReturnMapperBlock = new Resolver(
          this.typeUtils, this.elementUtils, missionModel.typeRules).applyRules(
          new TypePattern.ClassPattern(ClassName.get(ValueMapper.class),
                                       List.of(new TypePattern.ClassPattern(ClassName.get(Unit.class), List.of()))));
```